### PR TITLE
Readme and Travis Coverage

### PR DESCRIPTION
**Pull request by nabobalis**
Hopefully it works and looks pretty

### CODE REVIEW

**Comment by DanRyanIrish** [comment]
@nabobalis what do the changes to  .codecov.yaml and .travis.yml do in this PR?

**Comment by nabobalis** [comment]
We enabled the codecov project which tell us where tests are missing for the code. But I need to tell the bot to be quiet otherwise it comments on each PR

File: README.rst [hunk 1]
```diff
@@ -1,14 +1,75 @@
...
+   :target: https://codecov.io/gh/sunpy/ndcube
+.. |Powered by NumFOCUS| image:: https://img.shields.io/badge/powered%20by-NumFOCUS-orange.svg?style=flat&colorA=E1523D&colorB=007D8A
+   :target: http://numfocus.org
 
+NDcube is a a open-source package for multi-dimensional contiguious and non-contiguious spatially aware arrays.
 
+Installation
+------------
+
+If you want to develop NDcube you will need to install from git.
```
Comment (Cadair) [inline README.rst]: @DanRyanIrish we should re-write this now we are about to have a release

Comment (nabobalis) [reply README.rst]: Just a bit about installing from PyPi, I can do it here if you would like

**Comment by DanRyanIrish** [comment]
Yes, let me go over the README and make some suggestions.

**Comment by Cadair** [comment]
we should probably do it in a separate PR

**Comment by DanRyanIrish** [comment]
A general comment relevant to the REAME regards package naming convention. The convention that seems to have organically grown up is the ```ndcube``` is the package (no capitals). This reduces (although obviously not entirely) the confusion between the package ```ndcube``` and the class ```NDCube```. Do we think we should change the package name for any reason? For example ```NdcubePy``` follows more established conventions but is very ugly. Or should we change the capitalization convention?  Either way, we should be consistent to whatever convention we decide on to reduce confusion.  It would be easier to stick with what we have but if there are really compelling reasons to change I'm open for discussion.

**Comment by nabobalis** [comment]
That was me assuming it followed how sunpy was done. SunPy in text and without the capitals in code.

File: README.rst (hunk 1 above)
Comment (DanRyanIrish) [reply README.rst]: You mean rewrite to explain how to install the stable version?  We could explain how to do both.  Stable first, and then bleeding edge.  How about:

The stable version of ndcube can be installed simply via the anaconda distribution:
```conda install ndcube # Or whatever the full install instructions are```
or via pip:
```pip install ndcube # Or whatever the full install instructions are```

To install the bleeding edge version, you will need to install from git.  The best way to do this is to create a new conda environment and install the git version of ndcube:
```
$ conda config --append channels conda-forge
$ conda create -n ndcube-dev python sunpy hypothesis pytest-mock
$ source activate ndcube-dev
$ git clone https://github.com/sunpy/ndcube.git ndcube-git
$ cd ndcube-git
$ pip install -e .
```
For detailed (general) installation instructions, see...

File: README.rst [hunk 2]
```diff
@@ -1,14 +1,75 @@
...
+    $ cd ndcube-git
+    $ pip install -e .
+
+For detailed (general) installation instructions, see the `installation guide`_ in
+the SunPy docs.
+
+Getting Help
+------------
+
+For more information or to ask questions about SunPy, check out:
```
Comment (DanRyanIrish) [inline README.rst]: "SunPy"  should be "ndcube"

**Comment by Cadair** [comment]
I think `ndcube` for package and `NDCube` for class.

**Comment by DanRyanIrish** [comment]
Agreed!

File: README.rst [hunk 3]
```diff
@@ -17,12 +17,19 @@ software impact| |Powered by NumFOCUS| |Powered by Sunpy|
...
+The stable version of ndcube can be installed simply via the anaconda distribution using the `conda-forge channel`:
+:: 
+    $ conda install --channel conda-forge ndcube
+or via pip:
+::
+    $ pip install ndcube
+
+If you want to to install the bleeding edge version or develop NDcube you will need to install from git.
 The best way to do this is to create a new conda environment and install the git
 version of SunPy in it:
```
Comment (DanRyanIrish) [inline README.rst]: Should "SunPy" not also be "ndcube" here?

**Comment by nabobalis** [comment]
Changes are made, should be good to go after a once more.

File: README.rst (hunk 3 above)
Comment (nabobalis) [reply README.rst]: damn it, thanks!

**Comment by DanRyanIrish** [comment]
I think the mention of "SunPy" here: https://github.com/sunpy/ndcube/pull/68/files/691a3d0bdd24e44e6782bf99f3f76910de689bcb..d88b9c6236b0685a1adaaf62251f693cbabaf76f#diff-88b99bb28683bd5b7e3a204826ead112R63
should also be "ndcube" or "SunPy or "ndcube".

**Comment by nabobalis** [comment]
that sunpy could be the org sunpy :P

**Comment by nabobalis** [comment]
Probably best to squash merge then you do merge it.

File: README.rst [hunk 4]
```diff
@@ -1,14 +1,82 @@
...
+
+    $ conda config --append channels conda-forge
+    $ conda create -n ndcube-dev python sunpy hypothesis pytest-mock
+    $ source activate ndcube-dev
+    $ git clone https://github.com/sunpy/ndcube.git ndcube-git
+    $ cd ndcube-git
+    $ pip install -e .
+
+For detailed (general) installation instructions, see the `installation guide`_ in
+the SunPy docs.
```
Comment (DanRyanIrish) [inline README.rst]: Are the SunPy docs sufficient to give users more info here?  Or do we need an ndcube installation guide?

Comment (nabobalis) [reply README.rst]: It gives an overview of general installing. Conda setup and the like. It also walks through git install of sunpy. 

I would guess eventually that would be replaced with a link to the ndcube docs for installing it down the road.

Comment (DanRyanIrish) [reply README.rst]: OK.  If we are happy to do that in the future and not now, then I approve these changes!

Comment (DanRyanIrish) [reply README.rst]: Pending Travis and Appveyor passing, of course ;)

Thanks for this PR @nabobalis !

Comment (nabobalis) [reply README.rst]: No problem!

**Pull request merged**